### ISSUE TITLE
Gradients are not preserved in `torch`

### ISSUE THREAD

**Issue by frazane** [frazane/scoringrules#36]
When using `asarray()` in the torch backend, if called on a pre-existing tensor it will delete its autograd history by default (see https://pytorch.org/docs/stable/generated/torch.asarray.html). Consequently, one cannot compute gradients on functions that include `asarray()`. 

We need to use `as_tensor()` instead, or use `asarray(*args, requires_grad=True)`.
